app/scripts/views/complete_sign_up.js

```diff
@@ -70,89 +66,114 @@ define(function (require, exports, module) {
+
+      const code = verificationInfo.get('code');
+      const options = {
+        reminder: verificationInfo.get('reminder'),
         service: this.relier.get('service')
       };
-      return this.user.completeAccountSignUp(this.getAccount(), code, options)
-          .fail((err) => {
-            if (MarketingEmailErrors.created(err)) {
-              // A basket error should not prevent the
-              // sign up verification from completing, nor
-              // should an error be displayed to the user.
-              // Log the error and nothing else.
-              this.logError(err);
-            } else {
-              throw err;
-            }
-          })
-          .then(() => {
-            this.logViewEvent('verification.success');
-            this.notifier.trigger('verification.success');
-
-            // Update the stored account data in case it was
-            // updated by verifySignUp.
-            var account = this.getAccount();
-            this.user.setAccount(account);
-            return this.invokeBrokerMethod('afterCompleteSignUp', account);
-          })
-          .then(() => {
-            var account = this.getAccount();
-
-            if (! this.relier.isDirectAccess()) {
-              this._navigateToVerifiedScreen();
-              return;
-            }
 
-            return account.isSignedIn()
-              .then((isSignedIn) => {
-                if (isSignedIn) {
-                  this.navigate('settings', {
-                    success: t('Account verified successfully')
-                  });
-                } else {
-                  this._navigateToVerifiedScreen();
-                }
+      return this.user.completeAccountSignUp(account, code, options)
+        .fail((err) => this._logAndAbsorbMarketingClientErrors(err))
+        .then(() => this._onVerificationComplete(account))
+        .fail((err) => this._onVerificationError(err));
+    },
+
+    _logAndAbsorbMarketingClientErrors (err) {
+      if (MarketingEmailErrors.created(err)) {
+        // A basket error should not prevent the
+        // sign up verification from completing, nor
+        // should an error be displayed to the user.
+        // Log the error and nothing else.
+        this.logError(err);
+      } else {
+        throw err;
+      }
+    },
+
+    _onVerificationComplete (account) {
+      this.logViewEvent('verification.success');
+      this.notifier.trigger('verification.success');
+
+      // Update the stored account data in case it was
+      // updated by completeAccountSignUp.
+      this.user.setAccount(account);
+      return this.invokeBrokerMethod('afterCompleteSignUp', account)
+        .then(() => this._navigateToNextScreen());
+    },
+
+    _navigateToNextScreen () {
+      var account = this.getAccount();
+      var relier = this.relier;
+
+      if (relier.isSync()) {
+        if (this.isInExperimentGroup('connectAnotherDevice', 'treatment')) {
```

Comment:
We have a way to force treatment/control behavior - the tests use it already - https://github.com/mozilla/fxa-content-server/pull/4370/files#diff-4b30e0b3783b9ee05813174ab98c3916R85

experiment PR - https://github.com/mozilla/fxa-content-experiments/pull/49